src/main/java/com/google/pubsub/kafka/source/CloudPubSubSourceTask.java

```diff
@@ -136,18 +130,24 @@ public void start(Map<String, String> props) {
         (Long)
             validatedProps.get(
                 CloudPubSubSourceConnector.CPS_STREAMING_PULL_MAX_MS_PER_ACK_EXTENSION);
-    if (gcpCredentialsFilePath != null) {
-      try {
-        gcpCredentialsProvider.loadFromFile(gcpCredentialsFilePath);
-      } catch (IOException e) {
-        throw new RuntimeException(e);
+    String credentialsPath =
+        (String) validatedProps.get(ConnectorUtils.GCP_CREDENTIALS_FILE_PATH_CONFIG);
+    String credentialsJson =
+        (String) validatedProps.get(ConnectorUtils.GCP_CREDENTIALS_JSON_CONFIG);
+    ConnectorCredentialsProvider gcpCredentialsProvider;
+    if (credentialsPath != null) {
+      if (credentialsJson != null) {
+        throw new IllegalArgumentException(
```

Comment:
Is there a reason not to use ConnectorCredentialsProvider.fromConfig here as well?

Reply:
Done